src/Integration/Aws/SimpleS3/src/SimpleS3Client.php

```diff
@@ -21,8 +26,130 @@ public function getUrl(string $bucket, string $key): string
         return $this->getEndpoint($uri, [], null);
     }
 
-    /*
-     * TODO Add get() and put() which supports large files
-     * TODO add has()
+    public function download(string $bucket, string $key): ResultStream
+    {
+        return $this->getObject(['Bucket' => $bucket, 'Key' => $key])->getBody();
+    }
+
+    public function has(string $bucket, string $key): bool
+    {
+        return $this->objectExists(['Bucket' => $bucket, 'Key' => $key])->isSuccess();
+    }
+
+    /**
+     * @param string|resource|callable|iterable $object
+     * @param array{
+     *   ACL?: \AsyncAws\S3\Enum\ObjectCannedACL::*,
+     *   CacheControl?: string,
+     *   ContentLength?: string,
+     *   ContentType?: string,
+     *   Metadata?: array<string, string>,
+     * } $options
      */
+    public function upload(string $bucket, string $key, $object, array $options = []): void
+    {
+        // Send upload requests with size 32 MB
+        $megabyte = 1024 * 1024;
+
+        $contentLength = $options['ContentLength'] ?? null;
+        $chunksInPart = max(64, ceil($contentLength / (10000 * 1024 * 1024))); // number of chunk per part
```

Comment:
Could you help me to understand this line?

Why `10000`?
This seams like a strange way to define how big the parts should be...

Reply:
you're right... 10000 is the maximum number of parts...
but that's too much..  we should find a better formula !

Reply:
I've updated this. I added a configuration option. 

I think this is good.